conftest.py

```diff
@@ -221,12 +273,49 @@ def _benchmark_time():
 
 @pytest.fixture(scope="function")
 def auto_benchmark_time(benchmark_time):
+    """Benchmark the wall clock time of an individual test.
+
+    This fixture automatically records the wall clock time it takes to
+    execute an individual test if run as part of a benchmark. Depending on the test's structure, this may
+    include setup or teardown code that should not be measured. For more
+    control, use the ``benchmark_time`` context manager fixture.
```

Comment:
It seems we use this fixture only once, in `test_default_cluster_spinup_time` I wonder if we should just not use it. It's out of the scope of this PR, but any thoughts?

Reply:
I don't really have any strong feelings here, but I think that you're right that we may want to drop this to avoid people using it. I guess in 99% of the cases we want people to use the `benchmark_all` or `small_client` fixtures`. This is work for a follow-up PR, but I think `small_client` could also use some improved documentation, in particular we may want to highlight that it already uses the `benchmark_all` fixture.